DMCompiler/DM/DMProc.cs

```diff
@@ -316,7 +317,15 @@ public void LoopJumpToStart(string loopLabel)
         }
 
         public void LoopEnd() {
-            AddLabel(_loopStack.Pop() + "_end");
+            if (_loopStack?.TryPop(out var pop) ?? false)
+            {
+                AddLabel(pop + "_end");
+            }
+            else
+            {
+                DMCompiler.Error(new CompilerError(Location.Internal, "Cannot pop empty loop stack"));
```

Comment:
This should use the proc's location, so you can at least tell what proc is causing it.
```suggestion
                DMCompiler.Error(new CompilerError(Location, "Cannot pop empty loop stack"));
```